### PR TITLE
EVG-2648 Repotracker triggered by push events should trigger all possible project refs

### PR DESCRIPTION
**Pull request by richardsamuels**

### CODE REVIEW

File: model/project_ref_test.go [hunk 1]
```diff
@@ -197,3 +197,61 @@ func TestFindOneProjectRefByRepoAndBranch(t *testing.T) {
 	assert.Contains(err.Error(), "found 2 project refs, when 1 was expected")
 	require.Nil(projectRef)
 }
+
+func TestFindOneProjectRefByRepo(t *testing.T) {
```
Comment (bsamek) [inline model/project_ref_test.go]: test is misnamed

File: rest/data/repotracker.go [hunk 2]
```diff
@@ -46,40 +46,55 @@ func (c *RepoTrackerConnector) TriggerRepotracker(q amboy.Queue, msgID string, e
...
+			unactionable = append(unactionable, refs[i].Identifier)
+			continue
+		}
+
+		if err := q.Put(units.NewRepotrackerJob(fmt.Sprintf("github-push-%s", msgID), refs[i].Identifier)); err != nil {
+			catcher.Add(errors.Errorf("failed to add repotracker job to queue for project: '%s'", refs[i].Identifier))
+			failed = append(failed, refs[i].Identifier)
+
+		} else {
+			actionable = append(actionable, refs[i].Identifier)
```
Comment (bsamek) [inline rest/data/repotracker.go]: I think you can unindent this and remove the else.

File: rest/data/repotracker.go [hunk 3]
```diff
@@ -93,9 +108,11 @@ func (c *MockRepoTrackerConnector) TriggerRepotracker(_ amboy.Queue, _ string, e
 	if err != nil {
 		return err
 	}
+	if len(branch) == 0 {
```
Comment (bsamek) [inline rest/data/repotracker.go]: what does this case cover?

**Comment by bsamek** [comment]
I think the PR is fine, but I don't understand why we are doing this. What is the reason for it?

**Comment by richardsamuels** [comment]
@bsamek closing as not needed for now.

File: model/project_ref_test.go (hunk 1 above)
Comment (richardsamuels) [reply model/project_ref_test.go]: this never should've been created.

File: rest/data/repotracker.go (hunk 2 above)
Comment (richardsamuels) [reply rest/data/repotracker.go]: I should've called that "succeeded"

File: rest/data/repotracker.go (hunk 3 above)
Comment (richardsamuels) [reply rest/data/repotracker.go]: if it's not a commit push (such as a tag push), branch will be empty

Comment (richardsamuels) [reply rest/data/repotracker.go]: These aren't errors, so i didn't return an error.

**Pull request merged**